specifications/cloud_assembly.md

```diff
@@ -0,0 +1,398 @@
+# Cloud Assembly Specification, Version 1.0
+
+The key words **MUST**, **MUST NOT**, **REQUIRED**, **SHALL**, **SHALL NOT**, **SHOULD**, **SHOULD NOT**,
+**RECOMMENDED**, **MAY**, and **OPTIONAL** in this document are to be interpreted as described in [RFC 2119] when they
+are spelled out in bold, capital letters (as they are shown here).
+
+## Introduction
+A *Cloud Assembly* is a self-contained document container designed to hold the components of *cloud applications*,
+including all the parts that are needed in order to deploy those to a *cloud* provider. This document is the
+specification of the *Cloud Assembly* format as well as requirements imposed on *Cloud Assemblers* and *Cloud Runtimes*.
+
+### Design Goals
+The design goals for the *Cloud Assembly Specification* are the following:
+* The *Cloud Assembly Specification* is extensible.
+* The *Cloud Assembly Specification* is cloud-agnostic.
+* The *Cloud Assembly Specification* is easy to implement and use.
+* The *Cloud Assembly Specification* supports authenticity and integrity guarantees.
+* A *Cloud Assembly* is self-contained, making deployments reproductible.
+
+## Specification
+A *Cloud Assembly* is a ZIP archive that **SHOULD** conform to the [ISO/IEC 21320-1:2015] *Document Container File*
+standard. Use of the `deflate` compression method is **RECOMMENDED** in order to minimize the size of the resulting
+file. *Cloud Assembly* files **SHOULD** use the `.cloud` extension in order to make them easier to recognize by users.
+
+Documents in the archive can be stored with any name and directory structure, however the following entries at the root
+of the archive are reserved for special use:
+* `manifest.json` **MUST** be present and contains the [manifest document](#manifest-document) for the *Cloud Assembly*.
+* `signature.asc`, when present, **MUST** contain the [digital signature](#digital-signature) of the *Cloud Assembly*.
+
+### Manifest Document
+The `manifest.json` file is the entry point of the *Cloud Assembly*. It **MUST** be a valid [JSON] document composed of
+a single `object` that conforms to the following schema:
+
+Key           |Type                 |Required|Description
+--------------|---------------------|:------:|-----------
+`schema`      |`string`             |Required|The schema for the document. **MUST** be `cloud-assembly/1.0`.
+`drops`       |`Map<ID,Drop>`       |Required|A mapping of [*Logical ID*](#logical-id) to [Drop](#drop).
+`missing`     |`Map<string,Missing>`|        |A mapping of context keys to [missing information](#missing).
+
+The [JSON] specification allows for keys to be specified multiple times in a given `object`. However, *Cloud Assembly*
```

Comment:
Sooo... is this a long way to say "do whatever JSON does" ?

Reply:
It says "JSON allows whatever, but you SHOULD NOT use that trait of it" - insisting that re-using the same key multiple times results in undefined behavior.